lib/stringToParts.js

```diff
@@ -1,46 +1,59 @@
 'use strict';
 
 module.exports = function stringToParts(str) {
+
+  if (str.indexOf('[') === -1) {
+    if (str.indexOf('.') !== -1) {
+      return str.split('.');
+    }
+    return [str];
+  }
+
   const result = [];
 
   let curPropertyName = '';
-  let state = 'DEFAULT';
-  for (let i = 0; i < str.length; ++i) {
+  let i = 0;
+  const len = str.length;
+  let state = 0;
+
+  for (i = 0; i < len; ++i) {
+    const char = str[i];
+
     // Fall back to treating as property name rather than bracket notation if
     // square brackets contains something other than a number.
-    if (state === 'IN_SQUARE_BRACKETS' && !/\d/.test(str[i]) && str[i] !== ']') {
-      state = 'DEFAULT';
+    if (state === 1 && char !== ']' && !/\d/.test(char)) {
```

Comment:
@Uzlopak Should we add a `numericalRegEx`  here instead of creating a new regex each iteration of the loop?

Reply:
Benchmarks did not show any perf impact, I think. But sure will modify and bench again.